src/move-node.js

```diff
@@ -17,6 +17,41 @@ const moveNode = async (toNode) => {
     const oldNodes = await moveShard.moveShard(shard, toNode);
     removedNodes.push(...oldNodes);
   }
+  return removedNodes;
+};
+
+const replaceNodeInClusteredCouch = async (toNode, shardMapJson) => {
+  const removedNodes = [];
+  if (!shardMapJson) {
+    throw new Error('Shard map JSON is required for multi-node migration');
+  }
+  const shardMap = JSON.parse(shardMapJson);
```

Comment:
So ... it's not clear why we have the user pass the shard map as json if we generate it in the script. 
The node map output is useful because the user can change it, to assign shards to whatever node they desire, but in this case, the shard map should not be edited! 
So, I think this function should get the shard map internally and not require it as a parameter.

Reply:
The limiting factor here is that once couch is launched on the new servers, no information about any of the databases can be retrieved before shards are "moved" to the new nodes and metadata has been updated. That's the reason the mapping has to be retrieved from the old install if you want to keep the same mapping on the new install and avoid redistributing data from scratch - which is what we want to avoid.

Reply:
I know, that's why I suggested we move the shards while the old couchdb is running, and then start the new couchdb over the new structure.